test/test_jit.py

```diff
@@ -7892,6 +7892,17 @@ def test_as_tensor_tensor_input(input):
         inp = torch.randn(3, 4)
         self.checkScript(test_as_tensor_tensor_input, (inp,))
 
+    def test_torch_tensor_dtype(self):
+        def foo(s: float):
+            return torch.tensor(s)
+
+        scripted_foo = torch.jit.script(foo)
+        self.assertEqual(scripted_foo(1.).dtype, foo(1.).dtype)
+        saved_dtype = torch.get_default_dtype()
+        torch.set_default_dtype(torch.float32)
```

Comment:
Better try with torch.float64, too, in case we finally switch the JIT tests to use torch.float as the default dtype.

Wait, was this bug what was blocking that switch? Because ideally these tests would use torch.float32 as the default dtype like regular PyTorch and the rest of our test suite.